File: ios/BasicQRScanner/AVFoundation/AVFoundationViewManager.m
//
//  AVFoundationViewManager.m
//  BasicQRApp
//
//  Created by Ananth Desai on 30/12/24.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(AVFoundationViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(onSuccessfulScan, RCTDirectEventBlock)
@end

File: ios/BasicQRScanner/AVFoundation/AVFoundationViewManager.swift
//
//  AVFoundationViewManager.swift
//  BasicQRApp
//
//  Created by Ananth Desai on 30/12/24.
//

import Foundation

@objc(AVFoundationViewManager)
class AVFoundationViewManager: RCTViewManager {
  private weak var scannerRef: AVFoundationView? = nil
  override class func requiresMainQueueSetup() -> Bool {
    false
  }
  
  override func view() -> UIView! {
    let ref = AVFoundationView()
    scannerRef = ref
    return ref
  }
}
